apollo-router/src/configuration/mod.rs

```diff
@@ -103,34 +177,40 @@ impl JsonSchema for Plugins {
         let plugins = plugins()
             .iter()
             .sorted_by_key(|(name, _)| *name)
-            .map(|(name, factory)| (name.to_string(), factory.create_schema(gen)))
-            .collect::<schemars::Map<String, Schema>>();
-        let plugins_refs = plugins
-            .keys()
-            .map(|name| {
-                Schema::Object(SchemaObject {
-                    object: Some(Box::new(ObjectValidation {
-                        required: Set::from([name.to_string()]),
-                        ..Default::default()
-                    })),
-                    ..Default::default()
-                })
+            .filter(|(name, _)| name.starts_with(APOLLO_PLUGIN_PREFIX))
+            .map(|(name, factory)| {
+                (
+                    name[APOLLO_PLUGIN_PREFIX.len()..].to_string(),
+                    factory.create_schema(gen),
+                )
             })
-            .collect::<Vec<_>>();
+            .collect::<schemars::Map<String, Schema>>();
+        gen_schema(plugins)
+    }
+}
 
-        let plugins_object = SchemaObject {
-            object: Some(Box::new(ObjectValidation {
-                properties: plugins,
-                ..Default::default()
-            })),
-            subschemas: Some(Box::new(SubschemaValidation {
-                any_of: Some(plugins_refs),
-                ..Default::default()
-            })),
-            ..Default::default()
-        };
+#[derive(Clone, Debug, Default, Deserialize, Serialize, TypedBuilder)]
+#[serde(transparent)]
+pub struct UserPlugins {
+    pub plugins: Map<String, Value>,
+}
 
-        Schema::Object(plugins_object)
+impl JsonSchema for UserPlugins {
+    fn schema_name() -> String {
+        stringify!(Plugins).to_string()
+    }
+
+    fn json_schema(gen: &mut SchemaGenerator) -> Schema {
+        // This is a manual implementation of Plugins schema to allow plugins that have been registered at
+        // compile time to be picked up.
+
+        let plugins = plugins()
+            .iter()
+            .sorted_by_key(|(name, _)| *name)
+            .filter(|(name, _)| !name.starts_with("apollo."))
```

Comment:
use `APOLLO_PLUGIN_PREFIX` here too to be consistent?

Reply:
Fixed